### PR TITLE
Package id command

### PR DESCRIPTION
**Pull request by memsharded**
Merged with develop to solve conflicts. Most of the changes are from there, can be skipped

Most changes are very minor:
  - renamed ``filter`` (builtin word) => ``package_filter``
  - Linting (like removing trailing ; cleaning up spaces, lower_case_vars instead of mixedCaseVars
  - Extracted the print-paths functionality from the printing function, it was getting too long
  - Renamed outputFolder => output_folder, so it exactly matches the command line argument (filter)
  - Minor refactors to test, mostly linting, but also removed the "new", just using the given conanfile
  - Removed settings to tests, unnecessary

I have checked the folder creation issue, and the thing is that I think that showing the paths should be opt-in. It is an information that most users will not want, and the output is a bit cluttered. So I have added a ``--paths`` arg. 

It is true that folders are being created, but it seems it should be that way. It will not be a problem, because conan use a trailing ``/1`` subfolder that is used precisely so it is not being created and package creation workflow works. But once you ask for the path and the path is shortened, that cannot change. If the folder doesn't exist, there is a chance that such short random folder would be used by another package. I know, very unfortunate event, but still a chance. So better create and empty folder, not used, than having such a problem. And the creation of folders in any case would be restricted to ``--paths`` usage, and can be further restricted to ``--only=export_folder,package_folder`` if we don't want the empty intermediate source and build folder to be created.

TL;DR; making it opt-in, no big issue, can be merged.

### CODE REVIEW

**Comment by memsharded** [comment]
So basically, changes only in:
- https://github.com/claasd/conan/pull/3/files#diff-d5c08e7e637a05b6976c7ff31ddd62ddL1
- https://github.com/claasd/conan/pull/3/files#diff-6e707457c42d8e479af13c4b87e5ae2eR47
